DALOptimizer/Program.cs

```diff
@@ -0,0 +1,382 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Windows.Forms;
+
+using System.Text.RegularExpressions;
+using ICSharpCode.NRefactory.CSharp;
+using ICSharpCode.NRefactory.CSharp.Refactoring;
+using ICSharpCode.NRefactory.CSharp.Resolver;
+using ICSharpCode.NRefactory.Editor;
+using ICSharpCode.NRefactory.PatternMatching;
+using ICSharpCode.NRefactory.TypeSystem;
+
+namespace DALOptimizer
+{
+    class Program
+    {
+        [STAThread]
+        public static void Main(string[] args)
+        {
+
+            OpenFileDialog opnfd = new OpenFileDialog();
+            opnfd.ShowDialog();
+            string flName = opnfd.FileName;
+            
+            if (string.IsNullOrWhiteSpace(flName))
+            {
+                MessageBox.Show("Please Select one file ");
+                return;
+            }
+            
+
+            MatchInvocation Mr = new MatchInvocation();
+            
+            Solution solution = new Solution(flName);
+            MatchExpr MtchExp = new MatchExpr();
+
+            // Capture patterns & store it in respective lists of Node patterns
+
+            foreach (var file in solution.AllFiles)
+            {
+
+/*                if (file.FileName != "D:\\DALProject\\DAL\\RecharegeReportDAL.cs")
+                {
+                    continue;
+                }
+                */
+
+                if (Path.GetFileName(file.FileName) == "DatabaseProcessing.cs")
+                {
+                    continue;
+                }
+                var astResolver = new CSharpAstResolver(file.Project.Compilation, file.SyntaxTree, file.UnresolvedTypeSystemForFile);
+                foreach (var invocation in file.SyntaxTree.Descendants.OfType<AstNode>())
+                {
+
+                    //trak all expression Statemens
+                    if (invocation.GetType().Name == "ExpressionStatement")
+                    {
+                        Mr.MatchExprStmt(invocation, file, astResolver);
+                        continue;
+                    }
+                    //catch clause
+                    if (invocation.GetType().Name == "CatchClause")
```

Comment:
var nodeTypes = ["CatchClause", "FieldDeclaration"];
foreach (var nodeType in nodeTypes){
if (invocation.GetType().Name == nodeType)
 +                    {
 +                        Mr.MatchExprStmt(invocation, file, astResolver);
 +                        continue;
 +                    }
}